test/test_rpc.py

```diff
@@ -123,35 +125,15 @@ def test_self_add(self):
         ):
             dist.rpc(self_worker_name, torch.add, args=(torch.ones(2, 2), 1))
 
-    def test_duplicated_names(self):
+    def test_reinit(self):
         store = dist.FileStore(self.file.name, self.world_size)
-        dist.init_process_group(backend="gloo", rank=self.rank,
-                                world_size=self.world_size, store=store)
-        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+        dist.init_model_parallel('worker%d' % self.rank, rpc_backend=BACKEND,
+                                 rank=self.rank, world_size=self.world_size,
+                                 store=store)
+        with self.assertRaisesRegex(RuntimeError, "RPC is already initialized"):
             dist.init_model_parallel("duplicated_name")
         dist.join_rpc()
 
-    def test_invalid_names(self):
-        store = dist.FileStore(self.file.name, self.world_size)
-        dist.init_process_group(backend="gloo", rank=self.rank,
-                                world_size=self.world_size, store=store)
-
-        with self.assertRaisesRegex(RuntimeError, "Worker name must match"):
-            dist.init_model_parallel("abc*")
-
-        with self.assertRaisesRegex(RuntimeError, "Worker name must match"):
-            dist.init_model_parallel(" ")
-
-        with self.assertRaisesRegex(RuntimeError, "must be non-empty"):
-            dist.init_model_parallel("")
-
-        # If the number in the message does not match, it is likely that the
-        # value of MAX_NAME_LEN in RPC WorkerId has changed.
-        with self.assertRaisesRegex(RuntimeError, "shorter than 128"):
-            dist.init_model_parallel("".join(["a" for _ in range(500)]))
-
-        dist.join_rpc()
-
```

Comment:
Is this test no longer valid/needed?

Reply:
If other backend does not throw the same error, let's make this test specific for process group backend.

Reply:
Restored